Sources/Player/PlaybackEngine/PlayerEngine.swift

```diff
@@ -253,6 +257,26 @@ final class PlayerEngine {
 		}
 	}
 
+	func resetOrUnload() {
+		if featureFlagProvider.shouldUseImprovedCaching() {
+			unload()
+		} else {
+			reset()
+		}
+	}
+
+	func unload() {
```

Comment:
[suggestion] I believe we have briefly discussed this already, regarding maybe adding some explicit comments on the difference between the unload and reset, which from first look it is not super clear.